cassandra-analytics-common/src/main/java/org/apache/cassandra/spark/data/partitioner/CassandraInstance.java

```diff
@@ -34,20 +36,20 @@ public class CassandraInstance implements TokenOwner, Serializable
     public static final CassandraInstance.Serializer SERIALIZER = new CassandraInstance.Serializer();
 
     private static final long serialVersionUID = 6767636627576239773L;
-    private final String token;
+    private final Set<String> tokens;
     private final String node;
     private final String dataCenter;
 
-    public CassandraInstance(String token, String node, String dataCenter)
+    public CassandraInstance(Set<String> tokens, String node, String dataCenter)
     {
-        this.token = token;
+        this.tokens = tokens;
```

Comment:
Let's validate non-null or empty here . i.e
```suggestion
        this.tokens = Objects.requireNotNull(tokens, "tokens cannot be null");
```